sdk/communication/AzureCommunication/Source/Authentication/CommunicationAuthenticationPolicy.swift

```diff
@@ -28,16 +28,31 @@
 import AzureCore
 #endif
 import Foundation
-
+/**
+ The Azure Communication Services authentication policy. Conforms to Azure Cores Authenticating protocol.
+ */
 public class CommunicationUserCredentialPolicy: Authenticating {
```

Comment:
You are making a good point, initially this was added because the authentication headers that we were adding where not "Bearer". We should remove it now @JoshuaLai do you mind filing this in our backlog? Regarding the second comment, we had a chat with @schaabs and to avoid developers thinking that `CommunicationUserCredential` can be used in other Azure services, we decided to purposely not conform to `TokenCredential`. Instead we can have an internal adapter that converts `CommunicationUserCredential` to a `TokenCredential` in order to use `BearerTokenCredentialPolicy` (See [this](https://github.com/Azure/azure-sdk-for-net-pr/blob/feature/communication-2020-07-09/sdk/communication/Shared/src/CommunicationClientOptions.cs#L31)) or we should make `BearerTokenCredentialPolicy` more generic to not only support `TokenCredential`, but also support initializing with just `GetToken` as a function passed to it.